### DISCUSSION TITLE
get all columns of a table

### DISCUSSION THREAD

**Discussion by y0zong** [PostgREST/postgrest#2731] [Q&A] [answered]
hi, I'm new to postgrest, I need to get all columns from a table and can't find any detail from document, so how to make this work with postgrest? and what about add/delete column with postgrest?

**Comment by gregwebs** [comment]
For the columns, you are trying to get those from the API rather than a tool directly connected to the DB? Maybe you would fine that from the OpenAPI definitions?

**Comment by wolfgangwalther** [comment]
> and what about add/delete column with postgrest?

PostgREST is not designed for [DDL](https://www.postgresql.org/docs/current/ddl.html).

**Comment by steve-chavez** [comment, marked as answer]
Can be done with [pg_meta](https://github.com/aquameta/meta) - this extension exposes a `column` view that can be inserted/deleted. If you isolate this on a schema plus give it the right privileges, it should be fine to expose through postgREST.
